h2/src/main/org/h2/mode/FunctionsPostgreSQL.java

```diff
@@ -221,6 +227,16 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             // Not implemented
             result = v1.convertTo(Value.VARCHAR);
             break;
+        case ARRAY_TO_STRING:
+            StringJoiner joiner = new StringJoiner(v1.getString());
+            ValueArray va = (ValueArray) v0.convertTo(TypeInfo.TYPE_ARRAY);
```

Comment:
Such conversion to untyped array will be rejected in the future and PostgreSQL doesn't really accept values of other data types here. You need to check `getValueType()` and throw `DbException.getInvalidValueException(…)` if passed value is not an array.

Reply:
added type check

Reply:
Please, remove `convertTo()` invocation.

Reply:
removed